File: cpp/sprint5/D/code.cpp
/** Comment it before submitting
struct Node {  
  int value;  
  const Node* left = nullptr;  
  const Node* right = nullptr;  
};  
**/

#include "solution_tree.h"
#include <cassert>

using namespace std;


bool Solution(const Node* root1, const Node* root2) {
    // Your code
    // “ヽ(´▽｀)ノ”
} 

void test() {

    Node node1({1, nullptr, nullptr});
    Node node2({2, nullptr, nullptr});
    Node node3({3, &node1, &node2});

    Node node4({1, nullptr, nullptr});
    Node node5({2, nullptr, nullptr});
    Node node6({3, &node4, &node5});
    assert(Solution(&node3, &node6));
}

File: go/sprint5/D/code.go
package main

/**
Comment it before submitting
type Node struct {  
	value  int  
	left   *Node  
	right  *Node  
}
**/

func Solution(root1 *Node, root2 *Node) bool {
    // Your code
    // “ヽ(´▽｀)ノ”
}

func test() {
	node1 := Node{1, nil, nil}
    node2 := Node{2, nil, nil}
    node3 := Node{3, &node1, &node2}
    node4 := Node{1, nil, nil}
    node5 := Node{2, nil, nil}
    node6 := Node{3, &node4, &node5}
	
	if (!Solution(&node3, &node6)) {
		panic("WA")
	}
}